PluginCore/ScintillaNet/ScintillaControl.cs

```diff
@@ -5902,14 +5903,11 @@ private void OnSmartIndent(ScintillaControl ctrl, int ch)
                         this.BeginUndoAction();
                         try
                         {
-                            int position = CurrentPos;
-                            int curLine = LineFromPosition(position);
-                            int previousIndent = GetLineIndentation(curLine - 1);
-                            int match = SafeBraceMatch(position - 1);
-                            if (match != -1)
+                            int position = CurrentPos - 1;
+                            int match = SafeBraceMatch(position); //SafeBraceMatch() calls Colourise(0, -1)
+                            if (match != -1 && !PositionIsInString(position))
                             {
-                                previousIndent = GetLineIndentation(LineFromPosition(match));
-                                IndentLine(curLine, previousIndent);
+                                IndentLine(LineFromPosition(position), GetLineIndentation(LineFromPosition(match)));
```

Comment:
You are now doing `CurrentPos - 1`, so the line to indent may be the previous one.